### PR TITLE
[YUNIKORN-2575] Clarify error returned by IsPodFitNode

### PR DESCRIPTION
**Pull request by wilfred-s**
### What is this PR for?
IsPodFitNode should clearly set the error if the pod or node is missing from the cache when run. The result, if it is an error, contains the predicate plugin name that failed to improve identification in the events.

### What type of PR is it?
* [X] - Improvement

### What is the Jira issue?
* https://issues.apache.org/jira/browse/YUNIKORN-2575

### How should this be tested?
NA, no behaviour change just cleanup of return values

### CODE REVIEW

**Review by craigcondit — APPROVED** [review]
+1 LGTM (pending e2e test runs).

File: pkg/cache/context.go [hunk 1]
```diff
@@ -633,21 +639,28 @@ func (ctx *Context) EventsToRegister(queueingHintFn framework.QueueingHintFn) []
...
-		if targetNode := ctx.schedulerCache.GetNode(node); targetNode != nil {
-			// need to lock cache here as predicates need a stable view into the cache
-			ctx.schedulerCache.LockForReads()
-			defer ctx.schedulerCache.UnlockForReads()
-			_, err := ctx.predManager.Predicates(pod, targetNode, allocate)
-			return err
-		}
+	var pod *v1.Pod
+	var ok bool
+	if pod, ok = ctx.schedulerCache.GetPod(name); !ok {
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply pkg/cache/context.go:648]: Necessary due to structural differences in the K8s objects.

File: pkg/cache/context.go [hunk 2]
```diff
@@ -633,21 +639,28 @@ func (ctx *Context) EventsToRegister(queueingHintFn framework.QueueingHintFn) []
...
+		return ErrorPodNotFound
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply pkg/cache/context.go:649]: No, it is already logged in the caller, so this just becomes redundant.

File: pkg/cache/context.go [hunk 3]
```diff
@@ -633,21 +639,28 @@ func (ctx *Context) EventsToRegister(queueingHintFn framework.QueueingHintFn) []
...
+	}
+	// if pod exists in cache, try to run predicates
+	targetNode := ctx.schedulerCache.GetNode(node)
+	if targetNode == nil {
+		return ErrorNodeNotFound
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply pkg/cache/context.go:654]: No, it is already logged in the caller, so this just becomes redundant.

[the remark this answers is not in the document]
Comment (wilfred-s) [reply pkg/cache/context.go:654]: https://github.com/apache/yunikorn-core/blob/master/pkg/scheduler/objects/node.go#L394-L402

File: pkg/cache/context.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (wilfred-s) [reply pkg/cache/context.go:649]: https://github.com/apache/yunikorn-core/blob/master/pkg/scheduler/objects/node.go#L394-L402

File: pkg/cache/context.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wilfred-s) [reply pkg/cache/context.go:648]: Logged [YUNIKORN-2578](https://issues.apache.org/jira/browse/YUNIKORN-2578) for that as I noticed the same. Also logged [YUNIKORN-2577](https://issues.apache.org/jira/browse/YUNIKORN-2577) to clean up the IsPodFitNodeViaPreemption function

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**